### PR TITLE
Minor test-suite tweaks/fixes, volume III

### PR DESCRIPTION
**Pull request by dmnks**
This is the final batch before #2559.

### CODE REVIEW

File: tests/atlocal.in [hunk 1]
```diff
@@ -27,7 +29,7 @@ else
     exit
 fi
 
-RPMTEST="@CMAKE_CURRENT_BINARY_DIR@/testing"
+RPMTEST=$RPMTREE
```
Comment (pmatilai) [inline tests/atlocal.in:32]: This is missing quotes, which might matter in some exotic cases.

Comment (pmatilai) [reply tests/atlocal.in:32]: That said, the quoting is likely inconsistent to begin with, this just caught my eye as the previous assignment had them.

File: tests/atlocal.in [hunk 2]
```diff
@@ -81,28 +83,33 @@ fi
...
 function setup_env()
 {
-    if [ -d testing ]; then
-        RPMTEST=${PWD}/testing
-        HOME=${RPMTEST}
+    if [ -d tree ]; then
+        RPMTEST=${PWD}/tree
         RPM_CONFIGDIR=${RPMTEST}/@RPM_CONFIGDIR@
-        export RPMTEST HOME RPM_CONFIGDIR
+        export RPMTEST RPM_CONFIGDIR
```
Comment (pmatilai) [inline tests/atlocal.in:89]: Not overriding $HOME is the kind of change that makes this :sheep: very nervous as long as we're running outside a container. Does overriding get in the way of something in the later stages, or is it supposed to be a cleanup kind of thing? If the latter, please leave it out for now at least.

File: tests/atlocal.in (hunk 1 above)
Comment (dmnks) [reply tests/atlocal.in:32]: Ugh, you're right. Not that I wasn't aware, but I'm kinda sloppy when it comes to quoting in my own changes. I just consider quotes too noisy to bother :facepalm: Which, arguably, isn't a good idea :smile: 

And yup, it's inconsistent already, but that's no excuse to continue that. I'll fix this, thanks!

File: tests/atlocal.in (hunk 2 above)
Comment (dmnks) [reply tests/atlocal.in:89]: Oh-oh, again, you're right. On one hand, we do override a bunch of paths like `_topdir` in `run()` but you can never be sure that some program/script in the whole chain won't actually look at `$HOME`.

Us overriding it here for fakechroot is basically the poor-man's containerization that we're getting rid of, so while it's not a problem when you run the whole thing in a container, we're not there *yet* with this PR, so better leave that out, indeed.

And yup, it was a cleanup change, no other reason, really :smile:

File: tests/atlocal.in (hunk 1 above)
Comment (pmatilai) [reply tests/atlocal.in:32]: Oh I know the feeling. Guess why they're so inconsistent to begin with? :laughing:

File: tests/atlocal.in [hunk 3]
```diff
@@ -81,28 +83,33 @@ fi
...
-    (unset RPM_CONFIGDIR RPM_POPTEXEC_PATH
-     cd ${RPMTEST} && \
-     FAKECHROOT_BASE="${RPMTEST}" fakechroot "$@"
-    )
+    local cmd=$1
+    shift
+    case $cmd in
+        exec)
+            (unset RPM_CONFIGDIR RPM_POPTEXEC_PATH
+             cd ${RPMTEST} && \
```
Comment (pmatilai) [inline tests/atlocal.in:100]: Here's another case on unquoted RPMTEST. For whatever reason, it seems to be semi-consistently RPMTEST that is being unquoted. Amusingly, it's quoted on the next line though :smile:

Comment (dmnks) [reply tests/atlocal.in:100]: Yup, same goes with the use of curly braces, actually... I know they're needed in some cases such as when you have adjacent text that could be interpreted as part of the variable name, but I mix and match both styles quite a lot here :smile: 

I'll fix the quotes though.

**Comment by dmnks** [comment]
Fixed up. Other quote inconsistencies are left for later :smile:

File: tests/atlocal.in (hunk 3 above)
Comment (dmnks) [reply tests/atlocal.in:101]: > That said, lets not make this an "oh no, stop ship" situation. Missing quotes only affect marginal use-cases (in my books, using spaces in development directory names equal shooting your own foot just to be sure). So don't go chasing every unquoted variable now, it's not a show-stopper at all.

Oh yup, same thoughts here... i.e. if you're spaces in your filesnames, you're already doing it wrong :smile:

Comment (dmnks) [reply tests/atlocal.in:101]: Hmm, I actually forgot to fix up *this* occurrence, but as we discussed, let's not focus on that now :smile:

**Comment by pmatilai** [comment]
Okay, showtime :grin:

**Pull request merged**